### PR TITLE
For mozilla-mobile/fenix#9413 - Workaround Android 8 to update layoutDirection on activity recreation

### PR DESCRIPTION
**Pull request by boek**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/support/locale/src/main/java/mozilla/components/support/locale/LocaleAwareAppCompatActivity.kt [hunk 1]
```diff
@@ -15,4 +17,15 @@ open class LocaleAwareAppCompatActivity : AppCompatActivity() {
...
         super.attachBaseContext(context)
     }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // Android 8 has a bug which doesn't change the layoutDirection on activity recreation.
+        // https://github.com/mozilla-mobile/fenix/issues/9413
+        // https://stackoverflow.com/questions/46296202/rtl-layout-bug-in-android-oreo#comment98890942_46298101
+        if (Build.VERSION.SDK_INT == Build.VERSION_CODES.O) {
```
Comment (Amejia481) [inline components/support/locale/src/main/java/mozilla/components/support/locale/LocaleAwareAppCompatActivity.kt:27]: It looks good, lets just add a test for it :)

Comment (boek) [reply components/support/locale/src/main/java/mozilla/components/support/locale/LocaleAwareAppCompatActivity.kt:27]: 🤔 I started to go down that route. Do we have a place where we test build specific things already? I'm not sure how to do that on Android

Comment (Amejia481) [reply components/support/locale/src/main/java/mozilla/components/support/locale/LocaleAwareAppCompatActivity.kt:27]: [Robolectric allows to customized](http://robolectric.org/configuring/) the version of Android using  `@Config(sdk =[version])` :) 

Throughout the source code, we use it in a couple of places take a look at [SecureAbove22PreferencesTest]( https://github.com/mozilla-mobile/android-components/blob/afb265c3b7aff459066b041b12cc56068df773c0/components/lib/dataprotect/src/test/java/mozilla/components/lib/dataprotect/SecureAbove22PreferencesTest.kt#L20) 

If you need any additional help just let me know :)

**Pull request merged**